R/utils.R

```diff
@@ -98,19 +98,39 @@
 
     if (is.data.frame(tmp_response)) {
         offset <- self$MAX_API_LIMIT
+        addtl_responses <- list()
 
         while (nrow(tmp_response) == self$MAX_API_LIMIT) {
             query$offset <- offset
             tmp_response <- .single_request(self, url, query)
+            tmp_response <- .cast_lists_to_vectors(tmp_response)
 
-            response <- response %>% dplyr::bind_rows(tmp_response)
+            addtl_responses <- append(addtl_responses, list(tmp_response))
             offset <- offset + self$MAX_API_LIMIT
         }
+
+        if (length(addtl_responses) > 0) {
+            response <- data.table::rbindlist(
+                append(list(response), addtl_responses),
+                fill = TRUE
+            )
+        }
     }
 
     return(response)
 }
 
+.cast_lists_to_vectors <- function(df) {
```

Comment:
Is this function used yet?

Reply:
Yeah, on [line 106](https://github.com/American-Soccer-Analysis/itscalledsoccer-r/pull/10/files/355758c78255a517592fd34ff042f43d5cdfcb6a#diff-60bef5af43a1bd541299ce9bceeb226afb23deaa5e637b145c85e05ffd552b0cR106).

For context: I don't know if CRAN retains the check results, but on Fedora, some of the `team_id` values were inexplicably and all of a sudden being interpreted as lists (`list("a", "b")`) rather than vectors (`c("a", "b")`), which was preventing the unioning of those data frames.